sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -253,15 +263,53 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
         widgetTextureRenderer.render(widgetRenderer.getTexture())
       }
     } else {
+      logE("KIRYLDD", "render start $frameTimeNanos")
       mapboxRenderer.render()
+      logE("KIRYLDD", "render end $frameTimeNanos")
     }
 
     // assuming render event queue holds user's runnables with OpenGL ES commands
     // it makes sense to execute them after drawing a map but before swapping buffers
     // **note** this queue also holds snapshot tasks
     drainQueue(renderEventQueue)
+    if (hasViewAnnotations) {
+      hasViewAnnotations = false
+      if (viewAnnotationMode == ViewAnnotationUpdateMode.MAP_SYNCHRONIZED) {
+        Choreographer.getInstance().postFrameCallback {
+          swap(
+            nextFrameTimeNanos = it,
+            currentFrameTimeNanos = frameTimeNanos,
+            renderTimeNsCopy = renderTimeNsCopy,
+            expectedEndRenderTimeNanos = expectedEndRenderTimeNanos
+          )
+        }
+        return
+      }
+    }
+    // perform swap immediately if no view annotations are visible or mode is not MAP_SYNCHRONIZED
```

Comment:
Imho another comment is needed for `Choreographer.getInstance().postFrameCallback`, really worth to explain the subtle differences here

Also, can be simplified e.g.
`if (hasViewAnnotations && viewAnnotationMode == ViewAnnotationUpdateMode.MAP_SYNCHRONIZED) {} else {}`